lambda/js/src/test/scala/feral/lambda/ContextPlatformSuite.scala

```diff
@@ -0,0 +1,227 @@
+      generatedAwsRequestId <- Gen.alphaNumStr
+      generatedLogGroupName <- Gen.asciiPrintableStr
+      generatedLogStreamName <- Gen.asciiPrintableStr
+      generatedIdentity <- Gen.option(genCognitoIdentity).map(_.orUndefined)
+      generatedClientContext <- Gen.option(genClientContext).map(_.orUndefined)
+      generatedRemainingTimeInMillis <- Gen.posNum[Double]
+    } yield new facade.Context {
+      override def functionName: String = generatedFunctionName
+      override def functionVersion: String = generatedFunctionVersion
+      override def invokedFunctionArn: String = generatedInvokedFunctionArn
+      override def memoryLimitInMB: String = generatedMemoryLimitInMB
+      override def awsRequestId: String = generatedAwsRequestId
+      override def logGroupName: String = generatedLogGroupName
+      override def logStreamName: String = generatedLogStreamName
+      override def identity: js.UndefOr[facade.CognitoIdentity] = generatedIdentity
+      override def clientContext: js.UndefOr[facade.ClientContext] = generatedClientContext
+      override def getRemainingTimeInMillis(): Double = generatedRemainingTimeInMillis
+
+      override def toString(): String =
+        s"Context(functionName=$functionName, functionVersion=$functionVersion, invokedFunctionArn=$invokedFunctionArn, memoryLimitInMB=$memoryLimitInMB, awsRequestId=$awsRequestId, logGroupName=$logGroupName, logStreamName=$logStreamName, identity=$identity, clientContext=$clientContext, remainingTimeInMillis=${getRemainingTimeInMillis()}"
+    }
+  }
+
+  private implicit def optionCompare[A, B](
+      implicit C: Compare[A, B]): Compare[Option[A], Option[B]] =
+    new Compare[Option[A], Option[B]] {
+      override def isEqual(obtained: Option[A], expected: Option[B]): Boolean =
+        (obtained, expected).mapN(C.isEqual).getOrElse(obtained.isEmpty && expected.isEmpty)
+    }
+
+  private implicit val compareCognitoIdentity
+      : Compare[CognitoIdentity, facade.CognitoIdentity] =
+    new Compare[CognitoIdentity, facade.CognitoIdentity] {
+      override def isEqual(
+          obtained: CognitoIdentity,
+          expected: facade.CognitoIdentity): Boolean =
+        obtained.identityId == expected.cognitoIdentityId && obtained.identityPoolId == expected.cognitoIdentityPoolId
+    }
+
+  private implicit val compareClientContextClient
+      : Compare[ClientContextClient, facade.ClientContextClient] =
+    new Compare[ClientContextClient, facade.ClientContextClient] {
+      override def isEqual(
+          obtained: ClientContextClient,
+          expected: facade.ClientContextClient): Boolean =
+        obtained.installationId == expected.installationId &&
+          obtained.appTitle == expected.appTitle &&
+          obtained.appVersionName == expected.appVersionName &&
+          obtained.appVersionCode == expected.appVersionCode &&
+          obtained.appPackageName == expected.appPackageName
+    }
```

Comment:
This feels like a lot of boilerplate (which in theory could also have implementation mistakes), is it really worth it? 😅  versus converting one or two examples and just relying on case class equality.